lib/modem_info/modem_info_params.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <string.h>
+#include <stdlib.h>
+#include <cJSON.h>
+#include <modem_info.h>
+#include <at_params.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(modem_info_params);
+
+int modem_info_params_init(struct modem_param_info *modem)
+{
+	if (modem == NULL) {
+		return -EINVAL;
+	}
+
+	modem->network.cur_band.type	= MODEM_INFO_CUR_BAND;
+	modem->network.sup_band.type	= MODEM_INFO_SUP_BAND;
+	modem->network.area_code.type	= MODEM_INFO_AREA_CODE;
+	modem->network.operator.type	= MODEM_INFO_OPERATOR;
+	modem->network.mcc.type		= MODEM_INFO_MCC;
+	modem->network.mnc.type		= MODEM_INFO_MNC;
+	modem->network.cellid_hex.type	= MODEM_INFO_CELLID;
+	modem->network.ip_address.type	= MODEM_INFO_IP_ADDRESS;
+	modem->network.mode.type	= MODEM_INFO_MODE;
+	modem->network.system_mode.type	= MODEM_INFO_SYSTEMMODE;
+
+	modem->sim.uicc.type		= MODEM_INFO_UICC;
+	modem->sim.iccid.type		= MODEM_INFO_ICCID;
+
+	modem->device.modem_fw.type	= MODEM_INFO_FW_VERSION;
+	modem->device.battery.type	= MODEM_INFO_BATTERY;
+	modem->device.board		= CONFIG_BOARD;
+	modem->device.app_version	= STRINGIFY(APP_VERSION);
+
+	return 0;
+}
+
+static int check_system_mode(struct lte_param *system_mode)
+{
+	char lte_string[] = "LTE-M";
+	char nbiot_string[] = "NB-IoT";
+
+	if (system_mode == NULL) {
+		return -EINVAL;
+	}
+
+	if (system_mode->value == 1) {
+		memcpy(system_mode->string, lte_string,
+			strlen(lte_string));
+	} else if (system_mode->value == 0) {
+		memcpy(system_mode->string, nbiot_string,
+			strlen(nbiot_string));
+	} else {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int area_code_parse(struct lte_param *area_code)
+{
+	if (area_code == NULL) {
+		return -EINVAL;
+	}
+
+	area_code->string[2] = '\0';
+	area_code->value = (double)strtol(area_code->string, NULL, 16);
```

Comment:
A define for the `16` here would be nice, or a comment

Reply:
Added comment.